test/powershell/Modules/Microsoft.PowerShell.Management/Restart-Computer.Tests.ps1

```diff
@@ -0,0 +1,145 @@
+# note these will manipulate private data in the PowerShell engine which will
+# enable us to not actually restart the system, but return right before we do
+$TesthooksType = [system.management.automation.internal.internaltesthooks]
+$TesthookName = "TestStopComputer"
+$TesthookResultName = "TestStopComputerResults"
+$DefaultResultValue = 0
+function Set-HookResult([int]$result) {
+    ${TesthooksType}::SetTestHook($TesthookResultName, $result)
+}
+
+# protect the user from from restarting their computer 
+function Test-TesthookIsSet() {
+    try {
+        return ${TesthooksType}.GetField($TesthookName, "NonPublic,Static").GetValue($null)
+    }
+    catch {
+        # fall through
+    }
+    return $false
+}
+
+try 
+{
+    # set up for testing
+    $PSDefaultParameterValues["it:skip"] = ! $IsWindows
+    ${TesthooksType}::SetTestHook($TesthookName,$true)
+
+    Describe "Restart-Computer" -Tag Feature {
+        # if we throw in BeforeEach, the test will fail and the restart will not be called
+        BeforeEach {
+            if ( ! (Test-TesthookIsSet) ) {
+                throw "Testhook '${TesthookName}' is not set"
+            }
+        }
+
+        AfterEach {
+            Set-HookResult -result $defaultResultValue
+        }
+
+        It "Should restart the local computer" {
+            Set-HookResult -result $defaultResultValue
+            Restart-Computer -ea Stop| Should BeNullOrEmpty
+        }
+
+        It "Should support -computer parameter" {
+            Set-HookResult -result $defaultResultValue
+            Restart-Computer -computer (hostname) -ea Stop| Should BeNullOrEmpty

```

Comment:
-computer takes an array, we should perhaps use: hostname, localhost

Reply:
fixed